spec/glossary.adoc

```diff
@@ -2,12 +2,23 @@
 [[connected_mode]]Connected Mode:: The use of SonarLint associated with a SonarQube server or SonarCloud. See also <<standalone_mode,standalone mode>>.
 [[connection]]Connection:: A channel of communication between SonarLint and a particular SonarQube server or SonarCloud. See also <<connected_mode,connected mode>>.
 [[connection_config]]Connection Configuration:: All attributes that will be used to instantiate and operate a <<connection,connection>>. Usually provided by the IDE settings. It includes credentials, endpoint URL, timeouts, proxy...
+[[effective_rule_details]]Effective Rule Details:: <<rule_details>> taking into account the <<binding,binding>> and the <<finding_context,finding context>>. Some attributes can be different from the <<raw_rule_details, raw rule details>>.
+[[finding]]Finding:: Generic concept that includes all things reported by <<sonar_analyzer,Sonar analyzers>>: issues, security hotspots, taint vulnerabilities, ...
+[[finding_context]]Finding Context:: When reporting a finding, <<sonar_analyzer,Sonar analyzers>> can optionally provide a context key. This context key will be used to adapt the <<rule_details>> to give more specific remediation guidelines. For example, we can have framework specific context (spring vs struts).
 [[project_key]]Project Key:: Unique identifier of a <<sonar_project,Sonar project>> in a given SonarQube or SonarCloud instance. Project key can be changed on server side (the server use uuid internally).
+[[raw_rule_details]]Raw Rule Details:: Original <<rule_details>> as provided by <<sonar_analyzer>>.
+[[rule]]Rule:: A coding standard or practice which should be followed. <<sonar_analyzer,Sonar analyzers>> will analyze the user's code and report <<finding,findings>> for each rule. Each <<sonar_analyzer,Sonar analyzer>> will contribute a set of rules, identified by a unique <<rule_key,rule key>>. Rules have attached <<rule_details,details>> (name, type, tags, description, default activation...). Rules can also have parameters, with default values. On SonarQube or SonarCloud, users can create quality profiles, that are logical rule sets in which they can decide to activate some rules, and override certain rules metadata (severity, parameters, ...). <<sonar_project,Sonar projects>> are associated with a quality profile. When analyzing the code of a project, <<sensor,sensors>> are only interested in enabled rules and the effective parameter values. This is what we call <<active_rule,active rules>>.
+[[rule_details]]Rule Details:: All the attributes attached to a rule, like <<rule_key,key>>, name, severity, description, ... Some attributes can be overridden at different levels. For example severity and parameter values can be changed in the quality profile. The rule description can also be sensitive to the finding <<finding_context,context>>. See <<raw_rule_details>> and <<effective_rule_details>>.
```

Comment:
> The rule description can also be sensitive to the finding <<finding_context,context>>

Another idea that came to my mind, wouldn't we better introduce the concept of finding description ? The Rule description is what is provided by analyzers, with all contexts. The issue description is made from the rule description applied to a specific issue, that has a given context

Reply:
Maybe I am too much into technical implementation, but today the description is not attached to a finding. Multiple findings for the same rule key and context key will have the same description. So maybe this is too strong to call that "finding description"?